File: src/py/pe_0010.py
# SCRIPT


if __name__ == "__main__":
    # initialize variables
    size = 2000000
    half = size // 2
    prime = [i % 2 == 1 for i in range(size)]

    # one is not prime
    prime[1] = False
    gross = 2

    # disregard multiples of primes
    for i in range(3, half, 2):
        if prime[i]:
            gross += i
            for j in range(i * 2, size, i):
                prime[j] = False

    # continue the sum
    for i in range(i, size, 2):
        if prime[i]:
            gross += i

    # print out
    print(gross)

File: src/cpp/pe_0010.cpp
#include <iostream>
#include <vector>


// USING


using std::cout;
using std::endl;
using std::vector;


// SCRIPT


int main() {
    // initialize variables
    uint64_t i, j, size, half, gross;

    // array size
    size = 2000000;
    half = size / 2;

    // new array, all false
    vector<bool> prime;
    prime.resize(size, false);

    // two is prime, begin sum
    prime[2] = true;
    gross = 2;

    // all odds are potential primes
    for (i = 3; i < size; i += 2) {
        prime[i] = true;
    }

    // disregard multiples of primes
    for (i = 3; i < half; i += 2) {
        if (prime[i] == true) {
            gross += i;
            for (j = i * 2; j < size; j += i) {
                prime[j] = false;
            }
        }
    }

    // continue the sum
    for (; i < size; i += 2) {
        if (prime[i] == true) {
            gross += i;
        }
    }

    // print out
    cout << gross << endl;
}
